runners/samza/src/main/java/org/apache/beam/runners/samza/translation/PortableTranslationContext.java

```diff
@@ -48,26 +42,16 @@
   "rawtypes", // TODO(https://github.com/apache/beam/issues/20447)
   "nullness" // TODO(https://github.com/apache/beam/issues/20497)
 })
-public class PortableTranslationContext {
+public class PortableTranslationContext extends TranslationContext {
   private final Map<String, MessageStream<?>> messageStreams = new HashMap<>();
-  private final StreamApplicationDescriptor appDescriptor;
   private final JobInfo jobInfo;
-  private final SamzaPipelineOptions options;
-  private final Set<String> registeredInputStreams = new HashSet<>();
-  private final Map<String, Table> registeredTables = new HashMap<>();
-  private final HashIdGenerator idGenerator = new HashIdGenerator();
 
   private PipelineNode.PTransformNode currentTransform;
 
   public PortableTranslationContext(
       StreamApplicationDescriptor appDescriptor, SamzaPipelineOptions options, JobInfo jobInfo) {
+    super(appDescriptor, Collections.emptyMap(), options);
```

Comment:
do we not need the idMap for the portable context? It seems that we are using it in `TranslationContext#getIdForPValue()`

Reply:
We don't need it for now as it is only used in classic translation and side inputs in classic mode. The id in case of portable translation is available as part of protobuf representation of the transform.